src/GraphQL/GraphQLBuilderExtensions.cs

```diff
@@ -88,6 +88,56 @@ public static IGraphQLBuilder Configure<TOptions>(this IGraphQLBuilder builder,
             => builder.Configure<TOptions>(action == null ? null : (opt, _) => action(opt));
         #endregion
 
+        #region - RegisterAsBoth and TryRegisterAsBoth -
+        /// <summary>
+        /// Calls Register for both the implementation and service
+        /// </summary>
+        private static IGraphQLBuilder RegisterAsBoth<TService, TImplementation>(this IGraphQLBuilder builder, ServiceLifetime serviceLifetime)
+            where TService : class
+            where TImplementation : class, TService
+            => builder.Register<TImplementation>(serviceLifetime).Register<TService, TImplementation>(serviceLifetime);
+
+        /// <summary>
+        /// Calls Register for both the implementation and service
+        /// </summary>
+        private static IGraphQLBuilder RegisterAsBoth<TService, TImplementation>(this IGraphQLBuilder builder, Func<IServiceProvider, TImplementation> implementationFactory, ServiceLifetime serviceLifetime)
+            where TService : class
+            where TImplementation : class, TService
+            => builder.Register(implementationFactory, serviceLifetime).Register<TService>(implementationFactory, serviceLifetime);
+
+        /// <summary>
+        /// Calls Register for both the implementation and service
+        /// </summary>
+        private static IGraphQLBuilder RegisterAsBoth<TService, TImplementation>(this IGraphQLBuilder builder, TImplementation implementationInstance)
+            where TService : class
+            where TImplementation : class, TService
+            => builder.Register(implementationInstance).Register<TService>(implementationInstance);
+
+        /// <summary>
+        /// Calls Register for the implementation and TryRegister for the service
+        /// </summary>
+        private static IGraphQLBuilder TryRegisterAsBoth<TService, TImplementation>(this IGraphQLBuilder builder, ServiceLifetime serviceLifetime)
+            where TService : class
+            where TImplementation : class, TService
+            => builder.Register<TImplementation>(serviceLifetime).TryRegister<TService, TImplementation>(serviceLifetime);
```

Comment:
```suggestion
            => builder.TryRegister<TImplementation>(serviceLifetime).TryRegister<TService, TImplementation>(serviceLifetime);
```

?

Reply:
No; please read xml comments and changed code.

Reply:
Note that these are private methods, so users cannot be confused as to why `TryRegisterAsBoth` does not actually perform `TryRegister` on both.

Reply:
We could name the methods `RegisterAndTryRegister` !!

Reply:
ok

Reply:
ok ... rename to `RegisterAndTryRegister` ?  Or ok to leave as-is ?

Reply:
whatever